docs/guide/tutorials/task-guide.md

```diff
@@ -0,0 +1,524 @@
+add_task # <Task: Add>
+```
+
+Here we have created a simple Prefect Task named "Add" which receives two inputs (called x and y), and returns a single output. Note that we had to _instantiate_ our custom Task class in order to reach a proper Prefect Task object.
+
+#### The `@task` decorator
+
+We can avoid the object-oriented boilerplate altogether by writing a custom Python function and using the `@task` decorator:
+```python
+from prefect import task
+
+@task(name="Add")
+def add_task(x, y):
+    return x + y
+
+add_task # <Task: Add>
+```
+
+Here we have created an equivalent Task to the one above - the `@task` decorator essentially assigns this function as the `run` method of the underlying Task class for us.  A few observations are in order:
+
+- the `@task` decorator _instantiates_ a Prefect Task automatically
+- all [Task initialization keywords](https://docs.prefect.io/api/unreleased/core/task.html#task-2) can be provided to the decorator
+- if you choose not to provide a name, the function name will be used as the Task name
+
+
+::: warning Call signatures cannot be arbitrary
+When implementing a Prefect Task, whether via subclassing or the decorator, there _are_ some minor restrictions on your Task's call signature:
+
+First, all arguments to Prefect Tasks are ultimately treated as keyword arguments.  This means that arbitrary positional arguments (usually written `*args`) are _not_ allowed.
+
+In addition to disallowing `*args`, the following keywords are reserved and cannot be used by custom Prefect Tasks:
+- `upstream_tasks`: reserved for specifying upstream dependencies which do not exchange data
+- `mapped`: reserved for internal implementation details of mapping
+- `task_args`: reserved as a way of overriding task attributes when creating copies in the functional API
+- `flow`: reserved for internal implementation details
+
+If you violate any of these restrictions, an error will be thrown immediately at Task creation informing you.
+:::
+
+### Prefect Context contains useful information
+Sometimes it is useful to have additional information within your Task's runtime logic.  Such information is provided via Prefect's `Context` object.  `prefect.context` is a dictionary-like object (with attribute access for keys) containing useful runtime information.  For example, your Task's logger is stored in Prefect Context for adding custom logs to your task.
+```python
+import prefect
+
+@prefect.task
+def log_hello():
+    logger = prefect.context["logger"]
+    logger.info("Hello!")
+```
+Note that context is only populated _within a Flow run_.  This is important to be aware of when testing your task outside of a Flow run.  For a complete list of information available in Prefect Context, [see the API documentation](https://docs.prefect.io/api/unreleased/utilities/context.html).  For more information on how context works, see the associated [Concept Doc](https://docs.prefect.io/guide/core_concepts/execution.html#context).
```

Comment:
Perhaps a quick note that the context is a dictionary and can be gracefully accessed with the `.get` method?

Reply:
Good call, will update.